features/bootstrap/FeatureContext.php

```diff
@@ -0,0 +1,122 @@
+use Leviy\ReleaseTool\Versioning\SemanticVersioning;
+use Mockery\MockInterface;
+use PHPUnit\Framework\Assert;
+
+class FeatureContext implements Context
+{
+    /**
+     * @var MockInterface|VersionControlSystem
+     */
+    private $versionControlSystem;
+
+    /**
+     * @var ReleaseManager
+     */
+    private $releaseManager;
+
+    /**
+     * @var string|null
+     */
+    private $nextVersion;
+
+    /**
+     * @var MockInterface|InformationCollector
+     */
+    private $informationCollector;
+
+    public function __construct()
+    {
+        $this->informationCollector = Mockery::mock(InformationCollector::class);
+        $this->versionControlSystem = Mockery::mock(VersionControlSystem::class);
+        $this->versionControlSystem->shouldIgnoreMissing();
+
+        $changelogGenerator = Mockery::mock(ChangelogGenerator::class);
+        $changelogGenerator->shouldReceive('getChanges')->andReturn([]);
+
+        $this->releaseManager = new ReleaseManager(
+            $this->versionControlSystem,
+            new SemanticVersioning(),
+            $changelogGenerator,
+            []
+        );
+    }
+
+    /**
+     * @Given a release on this branch with version :version
+     */
+    public function aReleaseOnThisBranchWithVersion(string $version): void
+    {
+        $this->versionControlSystem->shouldReceive('getLastVersion')->andReturn($version);
+    }
+
+    /**
+     * @When I release a new :type version
+     */
+    public function iReleaseANewVersion(string $type): void
+    {
+        $this->selectVersionType($type);
+
+        $this->nextVersion = $this->releaseManager->determineNextVersion($this->informationCollector);
+    }
+
+    /**
+     * @When I release a(n) :preReleaseType version
+     * @When /^I release a (release candidate)$/
+     * @When I release a(n) :preReleaseType version of a new :type release
+     */
+    public function iReleaseAPreReleaseVersion(string $preReleaseType, ?string $type = null): void
+    {
+        if ($type !== null) {
+            $this->selectVersionType($type);
+        }
+
+        switch ($preReleaseType) {
+            case 'alpha':
+                $answer = 'a';
+                break;
+            case 'beta':
+                $answer = 'b';
+                break;
+            case 'release candidate':
+                $answer = 'rc';
+                break;
+        }
+
+        $this->informationCollector->shouldReceive('askMultipleChoice')->andReturn($answer);
+        $this->nextVersion = $this->releaseManager->determineNextPreReleaseVersion($this->informationCollector);
+    }
+
+    /**
+     * @Then version :version should be released
+     */
+    public function versionShouldBeReleased(string $version): void
+    {
+        Assert::assertSame($version, $this->nextVersion);
+    }
+
+    /**
+     * @return void
+     */
+    private function selectVersionType(string $type): void
+    {
+        if ($type === 'major') {
```

Comment:
I am always more fond of a switch-statement with these kinds of if-elseif-elseif-elseif-else-statements. Just taste-wise.